compiler/rustc_session/src/errors.rs

```diff
@@ -291,6 +291,23 @@ pub fn report_lit_error(sess: &ParseSess, err: LitError, lit: token::Lit, span:
         s.len() > 1 && s.starts_with(first_chars) && s[1..].chars().all(|c| c.is_ascii_digit())
     }
 
+    // Check if a prefix and suffix look correct irrespective of base.
+    fn looks_like_base_prefixed_number(prefix: &str, suffix: &str) -> bool {
+        let base = match suffix.chars().next().unwrap() {
```

Comment:
You can avoid the repeated `suffix.chars()` in this function. Do `let chars = suffix.chars()`, then do `chars.next().unwrap()` for the base check, then do `chars().filter(...)...` in the second part (thus avoiding the `skip(1)`).